src/java/org/apache/cassandra/service/accord/txn/TxnAppliedQuery.java

```diff
@@ -0,0 +1,156 @@
+
+// TODO: This is currently unused, but we might want to use it to support returning the condition result.
+public class TxnAppliedQuery implements Query
+{
+    private static final long EMPTY_SIZE = ObjectSizes.measure(new TxnAppliedQuery(ByteBufferUtil.EMPTY_BYTE_BUFFER));
+    public static class Applied implements Result
+    {
+        public static final Applied TRUE = new Applied();
+        public static final Applied FALSE = new Applied();
+        private static final long SIZE = ObjectSizes.measure(TRUE);
+
+        private Applied() {}
+
+        public boolean wasApplied()
+        {
+            return this == TRUE;
+        }
+
+        public static Applied valueOf(boolean b)
+        {
+            return b ? TRUE : FALSE;
+        }
+
+        public long estimatedSizeOnHeap()
+        {
+            return SIZE;
+        }
+
+        public static final IVersionedSerializer<Applied> serializer = new IVersionedSerializer<Applied>()
+        {
+            @Override
+            public void serialize(Applied applied, DataOutputPlus out, int version) throws IOException
+            {
+                out.writeBoolean(applied.wasApplied());
+            }
+
+            @Override
+            public Applied deserialize(DataInputPlus in, int version) throws IOException
+            {
+                return Applied.valueOf(in.readBoolean());
+            }
+
+            @Override
+            public long serializedSize(Applied applied, int version)
+            {
+                return TypeSizes.BOOL_SIZE;
+            }
+        };
+    }
+    private final ByteBuffer serializedCondition;
+
+    public TxnAppliedQuery(TxnCondition condition)
+    {
+        this.serializedCondition = serialize(condition, TxnCondition.serializer);
+    }
+
+    public TxnAppliedQuery(ByteBuffer serializedCondition)
+    {
+        this.serializedCondition = serializedCondition;
+    }
+
+    @Override
+    public boolean equals(Object o)
+    {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        TxnAppliedQuery query = (TxnAppliedQuery) o;
+        return Objects.equals(serializedCondition, query.serializedCondition);
+    }
+
+    @Override
+    public int hashCode()
+    {
+        return Objects.hash(serializedCondition);
+    }
+
+    @Override
+    public String toString()
+    {
+        return "TxnAppliedQuery{serializedCondition=" + deserialize(serializedCondition, TxnCondition.serializer) + '}';
+    }
+
+    public long estimatedSizeOnHeap()
+    {
+        return EMPTY_SIZE + ByteBufferUtil.estimatedSizeOnHeap(serializedCondition);
+    }
+
+    @Override
+    public Result compute(TxnId txnId, Data data, @Nullable Read read, @Nullable Update update)
+    {
+        TxnCondition condition = deserialize(serializedCondition, TxnCondition.serializer);
+        return condition.applies((TxnData) data) ? Applied.TRUE : Applied.FALSE;
```

Comment:
can the `Update` add the result?  If we will include in the returned `TxnData` then we could add a new TxnDataName.CONDITION_MET or something like that...  only concern with that is what happens when we add multi condition support